src/Microsoft.Health.Dicom.SqlServer/Features/ExtendedQueryTag/ExtendedQueryTagDataRowsBuilder.cs

```diff
@@ -45,12 +44,12 @@ internal static class ExtendedQueryTagDataRowsBuilder
             EnsureArg.IsNotNull(instance, nameof(instance));
             EnsureArg.IsNotNull(queryTags, nameof(queryTags));
 
-            var stringRows = new List<InsertStringExtendedQueryTagTableTypeV1Row>();
-            var longRows = new List<InsertLongExtendedQueryTagTableTypeV1Row>();
-            var doubleRows = new List<InsertDoubleExtendedQueryTagTableTypeV1Row>();
-            var dateTimeRows = new List<InsertDateTimeExtendedQueryTagTableTypeV1Row>();
-            var dateTimeWithUtcRows = new List<InsertDateTimeExtendedQueryTagTableTypeV2Row>();
-            var personNameRows = new List<InsertPersonNameExtendedQueryTagTableTypeV1Row>();
+            var stringRows = new HashSet<InsertStringExtendedQueryTagTableTypeV1Row>();
```

Comment:
Changing from List to HashSet here ensures that we only try to write one value to an ExtendedQueryTag* row, replacing the sequence-flattening procedure.